src/lib/ngc/compile-source-files.ts

```diff
@@ -35,27 +33,16 @@ export async function compileSourceFiles(
 
   // Don't use `ng.emit` as it doesn't output all errors.
   // https://github.com/angular/angular/issues/24024
+  const emitFlags = tsConfigOptions.declaration ? tsConfig.emitFlags : ng.EmitFlags.JS;
```

Comment:
Note: if there are not declarations to be emitted, we only want to emit the js files, this is due than when compiling in es5 the metadata and dts will be already emitted with the previous transpilation of es2015